node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -145,6 +147,10 @@ impl<C: Context> ActiveValidator<C> {
         instance_id: C::InstanceId,
         rng: &mut NodeRng,
     ) -> Vec<Effect<C>> {
+        let creator = state.unit(uhash).creator;
+        if creator == self.vidx {
+            return vec![Effect::DoppelgangerDetected(*uhash)];
+        }
```

Comment:
Isn't `on_new_unit` also called for our own units? (I guess not — otherwise the DES tests would fail now?)

Reply:
It's not, I run the network with `run-dev.sh` script and added a doppelganger validator.

Reply:
But the DES test did fail! :/ Probably some kind of programming error in the DES code.